### PR TITLE
Create separate 'autocomplete worker' for pyright

### PR DESCRIPTION
**Pull request by paked**

### CODE REVIEW

File: packages/runtime-python/package.json [hunk 1]
```diff
@@ -35,6 +43,7 @@
   "devDependencies": {
     "@rollup/plugin-node-resolve": "^15.0.1",
     "@rollup/plugin-typescript": "^9.0.2",
+    "comlink": "^4.4.1",
```
Comment (katharosada) [inline packages/runtime-python/package.json:46]: This being used yet?

File: packages/runtime-python/src/runtime/worker-manager.ts [hunk 2]
```diff
@@ -132,21 +156,22 @@ export class WorkerManager {
...
-    this.sendMessage({
+    this.sendMessageToRuntime({
       type: 'loadModule',
       moduleName: moduleName,
     })
   }
 
   sendParseTrees(parseTrees: ParseTrees) {
-    this.sendMessage({
+    console.log('sending parse trees')
```
Comment (katharosada) [inline packages/runtime-python/src/runtime/worker-manager.ts:166]: Remove log

File: packages/runtime-python/src/runtime/worker-manager.ts [hunk 3]
```diff
@@ -249,17 +274,46 @@ export class WorkerManager {
...
     this.stateCallBack(WorkerState.DISABLED)
-    this.worker.removeEventListener('message', this.handleMessageFromWorker)
-    this.worker.terminate()
-    this.worker = null
+
+    this.runtimeWorker.removeEventListener('message', this.handleMessageFromWorker)
+    this.runtimeWorker.terminate()
+    this.runtimeWorker = null
+
+    this.autocompleteWorker.removeEventListener('message', this.handleMessageFromAutocompleteWorker)
```
Comment (katharosada) [inline packages/runtime-python/src/runtime/worker-manager.ts:282]: We probably don't want to stop the autocomplete worker when we hit the stop button - that's half the reason we are separating it, no?

File: packages/runtime-python/src/runtime/worker-manager.ts [hunk 4]
```diff
@@ -249,17 +274,46 @@ export class WorkerManager {
...
+
     this.initialiseWorker()
   }
 
+  assessWorkerState() {
+    if (this.runtimeWorkerReady && this.autocompleteWorkerReady) {
+      console.log('autocomplete and runtime workers are ready')
+
+      this._workerState = WorkerState.READY
+      this.stateCallBack(WorkerState.READY)
```
Comment (katharosada) [inline packages/runtime-python/src/runtime/worker-manager.ts:294]: The state call back is only used for the worker - it's what makes the run button active. We probably want to say we're ready when the runtime worker is ready and keep the concept of 'ready' for the autcomplete worker separate.

File: packages/runtime-python/src/autocomplete_webworker.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  await tryModuleLoadPyodide()
+
+  pyodide = await setupPyodide([staticURLs.requestsPackageURL])
+
+  structuredProgram = createStructuredProgramWorker(new PyodideFakeFileSystem(typeshedPath, pyodide))
+
+  sendMessage({ type: 'ready' })
+}
+
+const updateParseTree = async (parseTreeInfo: ParseTreeInfo): Promise<SourceFile> => {
```
Comment (paked) [inline packages/runtime-python/src/autocomplete_webworker.ts:37]: copy pasta from webworker.ts

File: packages/runtime-python/src/runtime/autocomplete-worker-manager.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  sendParseTrees(parseTrees: ParseTrees) {
+    this.sendMessage({
+      type: 'parse_trees',
+      parseTrees,
+    })
+  }
+
+  requestExpressionTypeInfo(request: ExpressionTypeRequest) {
```
Comment (katharosada) [inline packages/runtime-python/src/runtime/autocomplete-worker-manager.ts:38]: There are going to be a lot of these methods, might be best to have a pass-this-through-to-worker method instead.

**Review by katharosada — APPROVED** [review]
LGTM - comment is optional

**Pull request merged**